enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,443 @@
+           - eipalloc-1234567890abcdef2
+           - eipalloc-1234567890abcdef3
+           - eipalloc-1234567890abcdef4
+           - eipalloc-1234567890abcdef5
+       type: AWS
+```
+3. Ingress Operator will create the default `IngressController` with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` with values from the field `eipAllocations`
+   from the cluster ingress config object.
+
+### API Extensions
+
+This proposal doesn't add any API extensions other than the new field proposed in
+[IngressConfig API Updates](#ingressconfig-api-updates).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+1. #### Set EIP through installer for the default IngressController:
+    1. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+    2. The installer then sets the ingress cluster object with the `eipAllocations`.
+    3. The cluster ingress operator then creates a default `IngressController` by:
+        1. creating an default `IngressController` CR.
+        2. then scaling a deployment for the `IngressController` CR which has the value of the `eipAllocations` from the
+           ingress cluster object.
+        3. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+           which uses the value from the field `eipAllocations` of `IngressController` CR.
+
+2. #### Validation on installer when installing in managed VPC (full-automated) based in the discovered zones used to create the cluster.
+     We will be comparing the number of Availability Zones in the region to the number of eipAllocations passed in the `install-config.yaml`.
+
+3. #### Validation on installer when installing in unmanaged (BYO VPC)
+     We will be comparing if the number of public subnets added to the install-config matches with len(eipAllocations).
+     However, The problem is that the AWS CCM can select subnets that aren't provided in the BYO Subnets, see https://issues.redhat.com/browse/OCPBUGS-17432.
+Here's one recommended option:
+
+The Installer should count all LB subnets by predicting what subnets be chosen by the AWS CCM 
+(i.e. any subnet without another cluster's kubernetes.io/cluster/<cluster-id> tag). 
+We can call this Predicted LB Subnet Count.
+
+We can examine the following scenarios:
+
+##### BYO Subnet Count != EIPs Allocations && BYO Subnet Count == Predicted LB Subnet count && Predicted LB Subnet count != EIPs Allocations:
```

Comment:
We might be able to eliminate all of this complex validation if:
- We agree to simplify the new subnets API (https://github.com/openshift/enhancements/pull/1634#discussion_r1814219682)
- EIP Allocations waits for new subnet API to be available and is dependent on it 

But it's a tough call. Maybe we have to do this validation in the beginning so you release EIP Allocations, but when the new subnets [API](https://github.com/openshift/enhancements/pull/1634) comes out, you can get rid of all of this complexity. I think that would be a massive win as far as a maintenance burden.

I would keep this validation here for now, but I will let you know if there are any updates.